Python/Product/PythonTools/PythonTools/Debugger/CustomDebuggerEventHandler.cs

```diff
@@ -83,5 +84,28 @@ public class CustomDebuggerEventHandler : IVsCustomDebuggerEventHandler110 {
                 "interpreter, please disable PGO.";
             MessageBox.Show(content, "PGO Is Not Supported", MessageBoxButton.OK, MessageBoxImage.Error);
         }
+
+        private void SetDebugOptions(IDebugEngine2 engine) {
+            var pyService = _serviceProvider.GetPythonToolsService();
+
+            var options = new StringBuilder();
+            if (pyService.DebuggerOptions.WaitOnAbnormalExit) {
+                options.Append(";" + AD7Engine.WaitOnAbnormalExitSetting + "=True");
+            }
+            if (pyService.DebuggerOptions.WaitOnNormalExit) {
+                options.Append(";" + AD7Engine.WaitOnNormalExitSetting + "=True");
+            }
+            if (pyService.DebuggerOptions.TeeStandardOutput) {
+                options.Append(";" + AD7Engine.RedirectOutputSetting + "=True");
+            }
+            if (pyService.DebuggerOptions.BreakOnSystemExitZero) {
+                options.Append(";" + AD7Engine.BreakSystemExitZero + "=True");
+            }
+            if (pyService.DebuggerOptions.DebugStdLib) {
+                options.Append(";" + AD7Engine.DebugStdLib + "=True");
+            }
+
+            engine.SetMetric(AD7Engine.DebugOptionsMetric, options.ToString());

```

Comment:
In the future I'd like to get back to revamping this whole thing such that all parts that deal with options (not just flags) or pass them around work with the same class that knows how to merge instances, stringify and parse. It's just too big of a change to take at this point. I'll leave a TODO comment along these lines.